Test collection fails with latest astropy main branch.

I just noticed that the latest commit to astropy main:  [d4f3216666718acd00bf41977b4439724bae3d7e](https://github.com/astropy/astropy/commit/d4f3216666718acd00bf41977b4439724bae3d7e), causes test collection to fail with the following error:
```python
===================================================================================================================== test session starts =====================================================================================================================
platform darwin -- Python 3.9.4, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
crds_context: jwst_0735.pmap
rootdir: /Users/wjamieson/projects/jwst, configfile: setup.cfg
plugins: cov-2.12.1, arraydiff-0.3, asdf-2.8.1, remotedata-0.3.2, requests-mock-1.9.3, xdist-2.3.0, ci-watson-0.5, filter-subpackage-0.1.1, hypothesis-6.14.0, openfiles-0.5.0, forked-1.3.0, astropy-header-0.1.2, doctestplus-0.9.0, jwst-1.2.4.dev11+gfc8a4077
collected 2444 items / 1 error / 2443 selected

=========================================================================================================================== ERRORS ============================================================================================================================
_________________________________________________________________________________________________ ERROR collecting jwst/associations/tests/test_level3_wfs.py _________________________________________________________________________________________________
jwst/associations/tests/test_level3_wfs.py:14: in <module>
    level3_asns = generate(pool, rules)
jwst/associations/generate.py:65: in generate
    existing_asns, new_asns, to_process = generate_from_item(
jwst/associations/generate.py:175: in generate_from_item
    new_asns, reprocess = rules.match(
jwst/associations/registry.py:157: in match
    asn, reprocess = rule.create(item, version_id)
jwst/associations/lib/dms_base.py:222: in create
    asn, reprocess = super(DMSBaseMixin, cls).create(item, version_id)
jwst/associations/association.py:139: in create
    matches, reprocess = asn.add(item)
jwst/associations/association.py:344: in add
    self._init_hook(item)
jwst/associations/lib/rules_level3.py:794: in _init_hook
    super(Asn_Lv3WFSCMB, self)._init_hook(item)
jwst/associations/lib/rules_level3_base.py:357: in _init_hook
    self.update_asn(item=item)
jwst/associations/lib/rules_level3_base.py:232: in update_asn
    self.data['target'] = self._get_target()
jwst/associations/lib/dms_base.py:627: in _get_target
    target_id = format_list(self.constraints['target'].found_values)
jwst/associations/lib/dms_base.py:734: in format_list
    return '-'.join(alist)
E   TypeError: sequence item 0: expected str instance, bytes found
=================================================================================================================== short test summary info ===================================================================================================================
ERROR jwst/associations/tests/test_level3_wfs.py - TypeError: sequence item 0: expected str instance, bytes found
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
====================================================================================================================== 1 error in 9.55s =======================================================================================================================
```
It appears as though this does not occur in any of the prior commits.

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -7,6 +7,12 @@ assign_wcs
 - Open the specwcs reference file for WFSS modes using the ``with`` context
   manger. [#6160]
 
+associations
+------------
+
+- Fix bug causing ``pytest`` to encounter an error in test collection when
+  running with recent commits to ``astropy`` main (``5.0.dev``). [#6176]
+
 cube_build
 ----------
 

--- a/jwst/associations/pool.py
+++ b/jwst/associations/pool.py
@@ -1,6 +1,9 @@
 """
 Association Pools
 """
+
+from pkg_resources import parse_version, get_distribution
+
 from astropy.io.ascii import convert_numpy
 
 from astropy.table import Table
@@ -50,7 +53,7 @@ class AssociationPool(Table):
         table = super(AssociationPool, cls).read(
             filename, delimiter=delimiter,
             format=format,
-            converters=_ConvertToStr(), **kwargs
+            converters=convert_to_str, **kwargs
         )
 
         # If anything has been masked, just fill
@@ -101,17 +104,27 @@ class AssociationPool(Table):
             )
 
 
+def _convert_to_str():
+    func, type_ = convert_numpy(str)
+
+    def convert_func(vals):
+        """Lowercase the conversion"""
+        results = func(vals)
+        results = [result.lower() for result in results]
+        return results
+
+    return [(convert_func, type_)]
+
+
 class _ConvertToStr(dict):
     def __getitem__(self, k):
-        func, type_ = convert_numpy(str)
-
-        def convert_func(vals):
-            """Lowercase the conversion"""
-            results = func(vals)
-            results = [result.lower() for result in results]
-            return results
-
-        return [(convert_func, type_)]
+        return _convert_to_str()
 
     def get(self, k, default=None):
         return self.__getitem__(k)
+
+
+if parse_version(get_distribution('astropy').version) >= parse_version('5.0.dev'):
+    convert_to_str = {'*': _convert_to_str()}
+else:
+    convert_to_str = _ConvertToStr()

